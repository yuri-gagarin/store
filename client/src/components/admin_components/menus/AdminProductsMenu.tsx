import React, { useState, useEffect, useRef } from "react";
import { Menu, MenuItemProps } from "semantic-ui-react";
// routing //
import { withRouter, RouteComponentProps, useRouteMatch } from "react-router-dom"
// css imports //
import "./css/adminProductsMenu.css";
// actions and state //
import { AppAction } from "../../../state/Store";

interface Props extends RouteComponentProps {
  dispatch: React.Dispatch<AppAction>
}
const AdminProductsMenu: React.FC<Props> = ({ history, location, dispatch }): JSX.Element => {
  const [ scrolled, setScrolled ] = useState<boolean>(false);
  const [ activeItem, setActiveItem ] = useState<string>("view_all");
  const [ menuOpen, setMenuOpen ] = useState<boolean>(false);

  const match = useRouteMatch("/admin/home/my_products");
  const adminProductsMenuRef = useRef<HTMLDivElement>(document.createElement("div"));

  const handleItemClick = (e: React.MouseEvent, { name }: MenuItemProps): void => {
    setActiveItem(String(name));


    switch (name) {
      case "view_all": {
        history.push(match?.path + "/all");
        break;
      }
      case "create": {
        history.push(match?.path + "/create");
        dispatch({ type: "CLEAR_CURRENT_PRODUCT", payload: null });
        break;
      }
      case "manage": {
        history.push(match?.path + "/manage");
        break;
      }
      case 'sorted': {
        history.push(match?.path + "/view_sorted");
        break;
      }
      default: history.push("/admin/home");
    }
  }
 
  const scrollListener = () => {
    if (window.scrollY > 1) {
      setScrolled(true);
    } else if (window.scrollY === 0) {
      setScrolled(false)
    } 
  };
  // lifecycle hooks //
  useEffect(() => {
    const currentURL = location.pathname;
    setTimeout(() => {
      setMenuOpen(true);
    }, 200);
    if (currentURL.match(/all/)) {
      setActiveItem("view_all");
    } else if (currentURL.match(/create/)) {
      setActiveItem("create");
    } else if (currentURL.match(/manage/)) {
      setActiveItem("manage");
    } else if (currentURL.match(/sorted/)) {
      setActiveItem("sorted");
    }
    // menu animation timeout //
    setTimeout(() => {
      setMenuOpen(true);
    }, 200);
  }, []);

  useEffect(() => {

    window.addEventListener("scroll", scrollListener, true);
    return () => window.removeEventListener("scroll", scrollListener);

  }, [adminProductsMenuRef]);

  return (
    <div className={ scrolled ? "adminProductsMenuFixed menuScrolled" : "adminProductsMenuFixed"} ref={adminProductsMenuRef}>
      <Menu tabular className={ menuOpen ? "adminProductsMenu productsMenuOpen" : "adminProductsMenu" }>
      <Menu.Item
          name='view_all'
          content="View All Products"
          active={activeItem === 'view_all'}
          onClick={handleItemClick}
        />
        <Menu.Item
          name='create'
          content="Create New Product"
          active={activeItem === 'create'}
          onClick={handleItemClick}
        />
        <Menu.Item
          name='manage'
          content="Manage Products"
          active={activeItem === 'manage'}
          onClick={handleItemClick}
        />
        <Menu.Item
          name="sorted"
          content="View Sorted Products"
          active={activeItem === "sorted"}
          onClick={handleItemClick}
        >
        </Menu.Item>
      </Menu>
    </div>
  )
};

export default withRouter(AdminProductsMenu);