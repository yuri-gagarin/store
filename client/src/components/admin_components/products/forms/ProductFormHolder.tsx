import React, { useState, useEffect, useContext } from "react";
import { Button, Grid } from "semantic-ui-react";
// client routing //
import { RouteComponentProps, withRouter } from "react-router-dom";
// css imports //
import "./css/productFormHolder.css";
// additional components //
import ProductForm from "./ProductForm";
import ProductImgPreviewHolder from "../image_preview/ProductImgPreviewHolder";
import LoadingBar from "../../miscelaneous/LoadingBar";
import ProductImgUplForm from "./ProductImgUplForm";
// state //
import { Store } from "../../../../state/Store";
// api and ui actions //
import { createProduct, editProduct } from "../actions/APIProductActions";
import { closeProductForm, openProductForm } from "../actions/UIProductActions";
// helpers //
import { ConvertDate } from "../../../helpers/displayHelpers";
import { checkSetValues } from "../../../helpers/validationHelpers";
// types 
import { FormState } from "./ProductForm";


interface Props extends RouteComponentProps {
 
}

type ProductData = {
  name: string;
  description: string;
  price: string;
  images: IProductImgData[];
}

const ProductFormHolder: React.FC<Props> = ({ history }): JSX.Element => {
  const { state,dispatch } = useContext(Store);
  const { loading, currentProductData, productFormOpen, error } = state.productState;
  const { name, description, price, createdAt, editedAt } = currentProductData;
  // local component state //
  // const [ formOpen, setFormOpen ] = useState<boolean>(false);
  const [ newForm, setNewForm ] = useState<boolean>(true);

  // ProductForm toggle //
  const toggleProductForm = () => {
    productFormOpen ? closeProductForm(dispatch) : openProductForm(dispatch);
  };
  // API call handlers CREATE - EDIT //
  const handleCreateProduct = ({ name, price, description }: FormState): void => {
    const productData: ProductData = {
      name,
      description,
      price,
      images: currentProductData.images
    };
    createProduct(productData, dispatch)
      .then((_) => {
        // product created //
        closeProductForm(dispatch);
      })
      .catch((_) => {
        // handle error or show modal //
      });
  };
  const handleUpdateProduct = ({ name, description, price }: FormState): void => {
    const productParams: ProductData = {
      name, description, price, images: currentProductData.images
    };
    editProduct(currentProductData._id, productParams, dispatch, state)
      .then((_) => {
        closeProductForm(dispatch)
      })
      .catch((_) => {
        // handle error show modal //
      });
  };
  // lifecycle hooks //
  useEffect(() => {
    if (!productFormOpen) {
      window.scrollTo({ top: 0, behavior: "smooth" });
    }
  }, [ productFormOpen ]);
  useEffect(() => {
    checkSetValues(currentProductData) ? setNewForm(false) : setNewForm(true);
  }, [ currentProductData ]);
  // component return //
  return (
    <div id="productFormHolder">
      { loading ? <LoadingBar /> : null }
      {
        !newForm ?
          <React.Fragment>
            <Grid.Row id={"productFormHolderDetails"}> 
              <Grid.Column mobile={16} tablet={14} computer={14}>
                <h1>Details</h1>
                <div className="productFormHolderDetailsItem">
                  <h3>Product name:</h3>
                  <p>{name}</p>
                </div>
                <div className="productFormHolderDetailsItem">
                  <h3>Product price:</h3>
                  <p>{price}</p>
                </div>
                <div className="productFormHolderDetailsItem">
                  <h3>Product description:</h3>
                  <p>{description}</p>
                </div>
                <div className="productFormHolderTimestamps">
                  <span>Created At: <strong>{ConvertDate.international(createdAt)}</strong></span>
                  <span>Edited At: <strong>{ConvertDate.international(editedAt)}</strong></span>
                </div>
              </Grid.Column>
            </Grid.Row>
            <Grid.Row>
              <Grid.Column mobile={16} tablet={14} computer={14}>
                <ProductImgPreviewHolder state={state} dispatch={dispatch} />
              </Grid.Column>
            </Grid.Row>
            <ProductImgUplForm />
          </React.Fragment>
          : null
      }
      <Grid.Row>
        <Grid.Column mobile={16} tablet={15} computer={14}>
          <Button  
            id="productFormToggleBtn"
            onClick={toggleProductForm} 
            content={ !productFormOpen ? "Open Form" : "Close Form"}
          />
          {
            productFormOpen ? 
              <ProductForm 
                name={name} 
                description={description} 
                price={price}
                handleCreateProduct={handleCreateProduct}
                handleUpdateProduct={handleUpdateProduct}
              /> 
            : null
          }
         
        </Grid.Column>
      </Grid.Row>
    </div>
  );
};
// export without router for tests //
export { ProductFormHolder };
// default export //
export default withRouter(ProductFormHolder);

