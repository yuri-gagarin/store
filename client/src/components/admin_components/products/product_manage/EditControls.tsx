import React from "react";
import { Button, Grid } from "semantic-ui-react";

interface Props {
  handleProductEdit(e:  React.MouseEvent<HTMLButtonElement>): void;
}

const CancelEditControls: React.FC<Props> = ({ handleProductEdit }): JSX.Element => {
  return (
    <Grid.Row>
      <Grid.Column>
         <Button content="Close Edit" onClick={handleProductEdit}></Button>
      </Grid.Column>
    </Grid.Row>
  );
};

export default CancelEditControls;
