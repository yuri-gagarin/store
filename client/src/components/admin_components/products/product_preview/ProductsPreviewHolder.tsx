import React, { useEffect, useState } from "react";
import { Grid, Item } from "semantic-ui-react";
// css imports //
import "./css/productsPreviewHolder.css";
// additional components //
import ProductPreview from "./ProductPreview";
import ProductsControls from "./ProductsControls";
import PopularProductsHolder from "./popular_products/PopularProductsHolder";
import LoadingScreen from "../../miscelaneous/LoadingScreen";
// types and interfaces //
import { AppAction, IGlobalAppState } from "../../../../state/Store";
// api actions //
import { getAllProducts } from "../actions/APIProductActions";

interface Props {
  state: IGlobalAppState;
  dispatch: React.Dispatch<AppAction>;
}

const ProductsPreviewHolder: React.FC<Props> = ({ state, dispatch }): JSX.Element => {
  const { loadedProducts } = state.productState;
  // local state //
  const [ dataLoaded, setDataLoaded ] = useState<boolean>(false);

  useEffect(() => {
    getAllProducts(dispatch)
      .then((success) => {
        if (success) {
          setDataLoaded(true);
        }
      })
  }, []);

  return (
    dataLoaded ? 
    <Grid stackable padded columns={2}>
      <Grid.Row>
      <Grid.Column computer={10} tablet={8} mobile={16}>
        <Item.Group>
          {
            loadedProducts.map((product) => {
              return (
                <ProductPreview 
                  key={product._id}
                  product={product}
                />
              );
            })
          }
        </Item.Group>
      </Grid.Column>
      <Grid.Column computer={6} tablet={8} mobile={16}>
        <ProductsControls totalProducts={loadedProducts.length} />
        <PopularProductsHolder popularProducts={loadedProducts}/>
      </Grid.Column>

      </Grid.Row>
      
    </Grid>
    :
    <LoadingScreen />
  )
};

export default ProductsPreviewHolder;