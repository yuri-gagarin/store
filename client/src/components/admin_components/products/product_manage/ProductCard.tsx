import React, { useState } from "react";
import { Button, Confirm, Grid } from "semantic-ui-react";
// additional components //
import EditControls from "./EditControls"
// css imports //
import "./css/productCard.css";
// actions and state /
import { deleteProduct } from "../actions/APIProductActions";
import { setCurrentProduct, clearCurrentProduct } from "../actions/UIProductActions";
import { IGlobalAppState, AppAction } from "../../../../state/Store";
// additional dependencies //
import { withRouter, RouteComponentProps } from "react-router-dom";
import { AdminProductRoutes } from "../../../../routes/adminRoutes";
// helpers //
import { ConvertDate } from "../../../helpers/displayHelpers";

interface Props extends RouteComponentProps {
  product: IProductData;
  imageCount: number;
  state: IGlobalAppState;
  dispatch: React.Dispatch<AppAction>;
}

const ProductCard: React.FC<Props> = ({ product, imageCount, state, dispatch, history }): JSX.Element => {
  const [ editing, setEditing ] = useState<boolean>(false);
  const [ confirmDeleteOpen, setConfirmDeleteOpen ] = useState<boolean>(false);

  const baseUrl = AdminProductRoutes.MANAGE_ROUTE;
  const { _id, name, price, description, images, createdAt, editedAt } = product;

  const handleProductOpen = (e: React.MouseEvent<HTMLButtonElement>): void => {
    history.push(baseUrl + "/view");
  }
  const handleProductEdit = (e: React.MouseEvent<HTMLButtonElement>): void => {
    if (!editing) {
      setCurrentProduct(_id, dispatch, state);
      history.push(baseUrl + "/edit");
      setEditing(true);
    } else {
      clearCurrentProduct(dispatch);
      history.push(baseUrl);
      setEditing(false);
    }
  };
  // delete product click and popup //
  const handleDeleteClick = (): void => {
    setConfirmDeleteOpen(true);
  };
  const confirmProductDeleteAction = (): void => {
    const { _id: productId } = product;
    deleteProduct(productId, dispatch, state)
      .then((_) => {
        // action successful //
      })
      .catch((_) => {
        // handle an error //
      });
  };
  const cancelStoreItemDeleteAction = (): void => {
    setConfirmDeleteOpen(false);
  };
 
  return (
    <React.Fragment>
      <Grid.Row style={{ padding: "0.5em", marginTop: "1em" }}>
        <Confirm 
          open={confirmDeleteOpen}
          onCancel={cancelStoreItemDeleteAction}
          onConfirm={confirmProductDeleteAction}
        />
        <div className="productManageDesc">
          <h3>Name: {name}</h3>
          <p><strong>Description:</strong> {description}</p>
          <span>Number of Images: <strong>{imageCount}</strong></span>
          <p><span>Created At: </span>{ConvertDate.international(createdAt)}</p>
        </div> 
        <div className="productManageCtrls">
          <Button 
            inverted
            color="green"
            className="productCardBtn" 
            content="Open" 
            onClick={handleProductOpen} 
          />
          <Button 
            inverted
            color="orange"
            className="productCardEditBtn" 
            content="Edit" 
            onClick={handleProductEdit}  
          />
          <Button 
            inverted
            color="red"
            className="productCardDeleteBtn" 
            content="Delete" 
            onClick={handleDeleteClick}
          />
        </div> 
      </Grid.Row>
      {  editing ?  <EditControls handleProductEdit={handleProductEdit}/>: null }
    </React.Fragment>
    
  );
};

export default withRouter(ProductCard);