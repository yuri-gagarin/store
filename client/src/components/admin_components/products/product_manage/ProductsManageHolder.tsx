import React, { useContext, useEffect, useRef, useState } from "react";
import { Button, Grid } from "semantic-ui-react";
// additional components //
import ProductFormHolder from "../forms/ProductFormContainer";
import ProductCard from "./ProductCard";
import LoadingScreen from "../../miscelaneous/LoadingScreen";
import ErrorScreen from "../../miscelaneous/ErrorScreen";
// actions and state //
import { getAllProducts } from "../actions/APIProductActions";
import { Store } from "../../../../state/Store";
// client routing //
import { withRouter, RouteComponentProps, useRouteMatch, Route } from "react-router-dom";
import { AdminProductRoutes } from "../../../../routes/adminRoutes";

interface Props extends RouteComponentProps {
};

const ProductsManageHolder: React.FC<Props> = ({ history }): JSX.Element => {
  const { state, dispatch } = useContext(Store);
  const { loading, loadedProducts, error } = state.productState;
  // local state //
  const [ newDataLoaded, setNewDataLoaded ] = useState<boolean>(false);
  const newProductsRef = useRef(loadedProducts);
  // routing //
  const match = useRouteMatch(AdminProductRoutes.MANAGE_ROUTE)
  const handleBack = () => {
    history.goBack();
  };

  // lifecycle hooks //
  useEffect(() => {
    let isMounted = true;
    if (isMounted) {
      getAllProducts(dispatch)
        .then((_) => {
          setNewDataLoaded(true);
        })
        .catch((_) => {
          setNewDataLoaded(false);
        });
    }
    return () => { isMounted = false };
  }, [ dispatch ]); 

  useEffect(() => {
    if (newProductsRef.current != loadedProducts && !loading && !error) {
      setNewDataLoaded(true);
    } 
  }, [ newProductsRef.current, loadedProducts, loading, error ]);
 
  return (
    newDataLoaded ?
    <Grid id="productsManageHolder" padded stackable columns={2}>
      <Route path={match?.url + "/edit"}> 
        <Grid.Row>
          <Grid.Column computer={12} tablet={6} mobile={16}>
            <h3>Editing Product: { state.productState.currentProductData.name }</h3>
            <Button 
              id="adminProductsManageBackBtn"
              inverted 
              color="green" 
              content="Back" 
              onClick={handleBack}>

            </Button>
          </Grid.Column>
        </Grid.Row>
        <ProductFormHolder />
      </Route>
      <Route exact path={match?.url}>
        <Grid.Row>
          <Grid.Column computer={12} tablet={8} mobile={16}>
          {
            loadedProducts.map((product) => {
              return (
                <ProductCard 
                  key={product._id}
                  product={product}
                  imageCount={product.images.length}
                  state={state}
                  dispatch={dispatch}
                />
              );
            })
          }
          </Grid.Column>
          <Grid.Column computer={4} tablet={8} mobile={16}>
            
          </Grid.Column>
        </Grid.Row>
      </Route>
    </Grid>
    : 
    ( 
      error ? <ErrorScreen lastRequest={ () => getAllProducts(dispatch) }/> : <LoadingScreen /> 
    )
  );
};

// export for testing //
export { ProductsManageHolder };
// default export //
export default withRouter(ProductsManageHolder);