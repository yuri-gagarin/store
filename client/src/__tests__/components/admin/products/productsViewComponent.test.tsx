import React, { ReactElement } from "react";
import { mount, ReactWrapper } from "enzyme";
import { Menu } from "semantic-ui-react";
// component imports //
import ProductsViewComponent from "../../../../components/admin_components/products/ProductsView";
import AdminProductsMenu from "../../../../components/admin_components/menus/AdminProductsMenu";
import ProductsPreviewHolder from "../../../../components/admin_components/products/product_preview/ProductsPreviewHolder";
import ProductFormHolder from "../../../../components/admin_components/products/forms/ProductFormHolder";
import { ProductsManageHolder } from "../../../../components/admin_components/products/product_manage/ProductsManageHolder";
// additional dependencies //
import { MemoryRouter as Router, Switch } from "react-router-dom";


describe("ProductView Component test", () => {
  let component: ReactWrapper;
  beforeAll(() => {
    component = mount<{}, typeof Router>(
      <Router initialEntries={["/admin/home/my_products"]}>
        <ProductsViewComponent />
      </Router>
    );
  });
  describe("ProductView Component render test", () => {
    it("Should correctly mount", () => {
      expect(component).toBeDefined();
    });
    it("Should render Admin Product Menu", () => {
      expect(component.find("AdminProductMenu")).toHaveLength(1);
    });
    it("Should render conditional routes", () => {
      expect(component.find("Switch")).toHaveLength(1);
    });
    it("Should have Render three children route componets", () => {
      const switchComponent = component.find(Switch);
      expect(switchComponent.props().children).toHaveLength(3);
    });
    it("Should have proper client routes and conditionally render correct Components", () => {
      const switchComponent = component.find(Switch);
      type RouteMap = {
        [key: string]: string
      }
      let map: RouteMap = {};

      const routeComponents = switchComponent.props().children as ReactElement[];
      for (const component of routeComponents) {
        if (component.props.children[1].type.WrappedComponent) {
          // to check if a component is wrapped in a WithRouter() function //
          map[component.props.path as string] = component.props.children[1].type.WrappedComponent;
        } else {
          map[component.props.path as string] = component.props.children[1].type;
        }
      }
      expect(map["/admin/home/my_products/all"]).toBe(ProductsPreviewHolder);
      expect(map["/admin/home/my_products/create"]).toBe(ProductFormHolder);
      expect(map["/admin/home/my_products/manage"]).toBe(ProductsManageHolder);
    });
  });
  describe("ProductView Component button actions", () => {
    describe("AdminProductMenu", () => {
      it("Should have 3 main navigation links", () => {
        const wrapper = component.find(AdminProductsMenu);
        const links = wrapper.find(Menu.Item)
        expect(links.length).toEqual(3);
      });
      it("'View All Products' link should properly function", () => {
        window.scrollTo = jest.fn();
        const viewAllLink = component.find(AdminProductsMenu).find(Menu.Item).at(0);
        viewAllLink.simulate("click");
        component.update();
        expect(component.find(ProductsPreviewHolder).length).toEqual(1);
        expect(component.find(ProductFormHolder).length).toEqual(0);
        expect(component.find(ProductsManageHolder).length).toEqual(0);
      });
      it("'Create Product' link should properly function", () => {
        window.scrollTo = jest.fn();
        const viewAllLink = component.find(AdminProductsMenu).find(Menu.Item).at(1);
        viewAllLink.simulate("click");
        component.update();
        expect(component.find(ProductsPreviewHolder).length).toEqual(0);
        expect(component.find(ProductFormHolder).length).toEqual(1);
        expect(component.find(ProductsManageHolder).length).toEqual(0);
      });
      it("'Manage Products' link should properly function", () => {
        window.scrollTo = jest.fn();
        const viewAllLink = component.find(AdminProductsMenu).find(Menu.Item).at(2);
        viewAllLink.simulate("click");
        component.update();
        expect(component.find(ProductsPreviewHolder).length).toEqual(0);
        expect(component.find(ProductFormHolder).length).toEqual(0);
        expect(component.find(ProductsManageHolder).length).toEqual(1);
      });
    });
    
  });
})