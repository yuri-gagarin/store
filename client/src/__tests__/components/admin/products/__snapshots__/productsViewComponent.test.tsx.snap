// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductView Component test ProductView Component render test Should Properly render Products View 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/admin/home/my_products",
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "34vlny",
            "pathname": "/admin/home/my_products",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "34vlny",
          "pathname": "/admin/home/my_products",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ProductsView>
      <div
        id="adminProductsViewHolder"
      >
        <withRouter(AdminProductsMenu)
          dispatch={[Function]}
        >
          <AdminProductsMenu
            dispatch={[Function]}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "canGo": [Function],
                "createHref": [Function],
                "entries": Array [
                  Object {
                    "hash": "",
                    "key": "34vlny",
                    "pathname": "/admin/home/my_products",
                    "search": "",
                    "state": undefined,
                  },
                ],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "index": 0,
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "key": "34vlny",
                  "pathname": "/admin/home/my_products",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            location={
              Object {
                "hash": "",
                "key": "34vlny",
                "pathname": "/admin/home/my_products",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
          >
            <div
              className="adminProductsMenuFixed"
            >
              <Menu
                className="adminProductsMenu"
                tabular={true}
              >
                <div
                  className="ui tabular adminProductsMenu menu"
                >
                  <MenuItem
                    active={true}
                    content="View All Products"
                    name="view_all"
                    onClick={[Function]}
                  >
                    <a
                      className="active item"
                      onClick={[Function]}
                    >
                      View All Products
                    </a>
                  </MenuItem>
                  <MenuItem
                    active={false}
                    content="Create New Product"
                    name="create"
                    onClick={[Function]}
                  >
                    <a
                      className="item"
                      onClick={[Function]}
                    >
                      Create New Product
                    </a>
                  </MenuItem>
                  <MenuItem
                    active={false}
                    content="Manage Products"
                    name="manage"
                    onClick={[Function]}
                  >
                    <a
                      className="item"
                      onClick={[Function]}
                    >
                      Manage Products
                    </a>
                  </MenuItem>
                  <MenuItem
                    active={false}
                    content="View Sorted Products"
                    name="sorted"
                    onClick={[Function]}
                  >
                    <a
                      className="item"
                      onClick={[Function]}
                    >
                      View Sorted Products
                    </a>
                  </MenuItem>
                </div>
              </Menu>
            </div>
          </AdminProductsMenu>
        </withRouter(AdminProductsMenu)>
        <Switch />
      </div>
    </ProductsView>
  </Router>
</MemoryRouter>
`;
